### DISCUSSION TITLE
HTTPException always returns 200 [QUESTION]

### DISCUSSION THREAD

**Discussion by galvakojis** [fastapi/fastapi#7444] [Questions]
Maybe I am missing something as usual
as i understand 
```
 raise HTTPException(
            status_code=status.HTTP_415_UNSUPPORTED_MEDIA_TYPE,
            detail="File is not excel",
            headers={"X-Error": "There goes my error"},

        ) 
```
should return status code of 415, but I just always get 200, if line is changed to 
`return Response(status_code=status.HTTP_415_UNSUPPORTED_MEDIA_TYPE, content="File is not excel",)`
voula I get a 415 response code, so I am a bit lost here. Maybe you can explain me, where is my mistake?

**Comment by Dustyposa** [comment]
Did you have `exception_handler` ?

**Comment by galvakojis** [comment]
No, I don't, but  I am not using `custom exception handlers`

https://fastapi.tiangolo.com/tutorial/handling-errors/#use-httpexception

**Comment by galvakojis** [comment]
got it, you need to use  `exception_handler`

so using this example is useless? 
![image](https://user-images.githubusercontent.com/17315578/83149602-f6bfe880-a102-11ea-84bf-e79d09849e30.png)

**Comment by tiangolo** [comment]
Thanks for the help here @Dustyposa ! :clap: :bow: 

I imagine you solved your problem, so thanks for closing the issue :+1:

**Comment by galvakojis** [comment]
Hello, well I think as 99% it happens to me me, it was user error, Maybe I forgot to import something or imported a wrong module
   On Tuesday, October 18, 2022 at 12:29:39 PM GMT+3, Saeed Esmaili ***@***.***> wrote:  
 
 


@galvakojis I didn't get how you fixed this issue. I'm having the same issue, and couldn't find anything about exception_handler in the docs.
@app.get("/ping")
async def pong(request: Request):
    return HTTPException(status_code=400, detail="oops")
curl returns:
> GET /ping HTTP/1.1
> Host: localhost:8002
> User-Agent: curl/7.79.1
> accept: application/json
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< date: Tue, 18 Oct 2022 09:25:19 GMT
< server: uvicorn
< content-length: 50
< content-type: application/json
<
* Connection #0 to host localhost left intact
{"status_code":400,"detail":"oops","headers":null}%
